Slim/Control/Request.pm

```diff
@@ -1528,7 +1528,8 @@ sub addParamPos {
 
 # get a parameter by name
 sub getParam {
-	return $_[0]->{'_params'}->{ $_[1] };
+	my $r = $_[0]->{'_params'}->{ $_[1] };
+	return ($r ne '') ? $r : undef;
```

Comment:
Why not return whatever was stored? Even if it was an empty string, a caller might actually expect a string. Changing this could cause a lot of issues. IMHO there's nothing wrong with the current implementation.

Reply:
I wish I had been more verbose when I created that patch. I've been using it for at least 8 years already.

Browsing the code I think the reason for changing it was probably because a construct like this:
```my $sort          = $request->getParam('sort') || 'title';```
would never return 'title' unless that was what sort actually stated.

In hindsight a better version would probably be:
``` return defined($_[0]->{'_params'}->{ $_[1] })? $_[0]->{'_params'}->{ $_[1] } : undef;```

Reply:
That `$sort` would very well become `title` if a string was returned. `undef`, `''`, or `0` are all falsy values in Perl, thus would fall back to `title` in the above case.

I see no reason to change this. It has been this way in almost two decades, does what I'd expect it to do. And nobody has complained so far (besides you, of course 😁).